ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/DivulgenceIT.scala

```diff
@@ -223,4 +230,42 @@ final class DivulgenceIT extends LedgerTestSuite {
       // nothing to test, if the workflow ends successfully the test is considered successful
     }
   })
+
+  test(
+    "DivulgenceDivulgeAfterArchival",
+    "Divulging, archiving and divulging again a contract with key should be possible",
+    allocate(SingleParty, SingleParty),
+  )(implicit ec => { case Participants(Participant(alpha, partyA), Participant(beta, partyB)) =>
+    for {
+      // Create a helper contract where partyB is a signatory
+      helper <- beta.create(
+        partyB,
+        WithKeyDivulgenceHelper(divulgedTo = partyB, withKeyOwner = partyA),
+      )
+
+      _ <- synchronize(alpha, beta)
+
+      // Create a WithKey contract owned by the partyA
+      withKey1 <- alpha.create(partyA, WithKey(partyA))
+
+      // Divulge the withKey1 contract
+      _ <- alpha.exercise(partyA, helper.exerciseWithKeyDivulgenceHelper_Fetch(_, withKey1))
+
+      _ <- synchronize(alpha, beta)
+
+      // Archive the withKey1 contract
+      _ <- alpha.exercise(partyA, withKey1.exerciseWithKey_Archive(_))
```

Comment:
any reason not to use the autogenerated `Archive` here, i.e., `exerciseArchive(_)`?

Reply:
I haven't noticed the autogenerated one. Thank you!

Reply:
I've changed the test case to use the autogenerated `Archive` and due to this I needed to restructure imports to avoid implicits ambiguity.